### PR TITLE
Add Mutating & Validating webhook for SampleSource

### LINKED ISSUE #52: Sample Source needs a webhook
The controller process can expose it, or it can run separately, but we should be pushing folks to have one.

cc @n3wscott

### PR DESCRIPTION
**Pull request by tom24d**
Fixes #52 .

basically, a shameless copy of GithubSource, partially from sample-controller and several comments to describe the behavior.

### CODE REVIEW

File: cmd/webhook/main.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func main() {
+	ctx := webhook.WithOptions(signals.NewContext(), webhook.Options{
+		ServiceName: "sample-source-webhook",
+		Port:        8443,
+		SecretName:  "webhook-certs",
+	})
+
+	sharedmain.WebhookMainWithContext(ctx, "sample-source-webhook",
```
Comment (lberk) [inline cmd/webhook/main.go:125]: could we const or var the `sample-source-webhook` please?

File: pkg/apis/samples/v1alpha1/samplesource_validation.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	//errs is nil if everything is fine.
+	return errs
+}
+
+// Validate validates SampleSourceSpec.
+func (sspec *SampleSourceSpec) Validate(ctx context.Context) *apis.FieldError {
+	//Add code for validation webhook for SampleSourceSpec
+	var errs *apis.FieldError
+
```
Comment (lberk) [inline pkg/apis/samples/v1alpha1/samplesource_validation.go:40]: given that this is a sample source, specifically for beginners, it'd be great if we could actually show validation of our simple spec fields

Comment (lberk) [reply pkg/apis/samples/v1alpha1/samplesource_validation.go:40]: (ie, not just the sink)

File: pkg/apis/samples/v1alpha1/samplesource_defaults.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+package v1alpha1
+
+import (
+	"context"
+)
+
+// SetDefaults mutates SmapleSource.
+func (s *SampleSource) SetDefaults(ctx context.Context) {
+	//Add code for Mutating admission webhook.
+
```
Comment (lberk) [inline pkg/apis/samples/v1alpha1/samplesource_defaults.go:26]: crazy idea: what if we made the `spec.interval` variable optional, but then set a default (say, 10s?) here to demonstrate the defaulting webhook as well?

File: pkg/apis/samples/v1alpha1/samplesource_validation.go (hunk 2 above)
Comment (tom24d) [reply pkg/apis/samples/v1alpha1/samplesource_validation.go:40]: OK. I am going to add validation for the other fields.

File: pkg/apis/samples/v1alpha1/samplesource_defaults.go (hunk 3 above)
Comment (tom24d) [reply pkg/apis/samples/v1alpha1/samplesource_defaults.go:26]: Right! I will add it also.

**Comment by mattmoor** [comment]
Can you pull in the latest pkg (`./hack/update-deps.sh --upgrade`) and add the block of yaml from: https://github.com/knative/pkg/pull/1318 to the process running the webhook.

**Comment by lberk** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**